### PR TITLE
Try better user flexibility when creating figures

### PR DESCRIPTION
**Pull request by MOchiara**

### CODE REVIEW

File: glidertest/tools.py [hunk 1]
```diff
@@ -488,15 +488,21 @@ def plot_section_with_srss(ds, ax, sel_var='TEMP',start_time = '2023-09-06', end
     plt.colorbar(c, label=f'{sel_var} [{ds[sel_var].units}]')
     return ax 
  
-def check_temporal_drift(ds, ax1, ax2, var='DOXY'):
-    ax1.scatter(mdates.date2num(ds.TIME),ds[var], s=10)
-    ax1.xaxis.set_major_formatter(DateFormatter('%Y-%m-%d'))
-    ax1.set(ylim=(np.nanpercentile(ds[var], 0.01), np.nanpercentile(ds[var], 99.99)), ylabel=var)
+def check_temporal_drift(ds: pd.DataFrame, var: str,ax: plt.Axes = None, **kw: dict,)-> tuple({plt.Figure, plt.Axes}):
```
Comment (callumrollo) [inline glidertest/tools.py:491]: conventionally, one would expect ds to be an xarray DataSet, we'd use df for a pandas DataFrame. Nice type hints though!

File: glidertest/tools.py [hunk 2]
```diff
@@ -488,15 +488,21 @@ def plot_section_with_srss(ds, ax, sel_var='TEMP',start_time = '2023-09-06', end
...
  
-def check_temporal_drift(ds, ax1, ax2, var='DOXY'):
-    ax1.scatter(mdates.date2num(ds.TIME),ds[var], s=10)
-    ax1.xaxis.set_major_formatter(DateFormatter('%Y-%m-%d'))
-    ax1.set(ylim=(np.nanpercentile(ds[var], 0.01), np.nanpercentile(ds[var], 99.99)), ylabel=var)
+def check_temporal_drift(ds: pd.DataFrame, var: str,ax: plt.Axes = None, **kw: dict,)-> tuple({plt.Figure, plt.Axes}):
+    if ax is None:
+        fig, ax = plt.subplots(1, 2, figsize=(14, 6))
+    else:
+        ax = ax
```
Comment (callumrollo) [inline glidertest/tools.py:495]: is ax=ax needed? Seems superfluous

File: glidertest/tools.py [hunk 3]
```diff
@@ -488,15 +488,21 @@ def plot_section_with_srss(ds, ax, sel_var='TEMP',start_time = '2023-09-06', end
...
     
-    c=ax2.scatter(ds[var],ds.DEPTH,c=mdates.date2num(ds.TIME), s=10)
-    ax2.set(xlim=(np.nanpercentile(ds[var], 0.01), np.nanpercentile(ds[var], 99.99)),ylabel='Depth (m)', xlabel=var)
-    ax2.invert_yaxis()
+    c=ax[1].scatter(ds[var],ds.DEPTH,c=mdates.date2num(ds.TIME), s=10)
+    ax[1].set(xlim=(np.nanpercentile(ds[var], 0.01), np.nanpercentile(ds[var], 99.99)),ylabel='Depth (m)', xlabel=var)
+    ax[1].invert_yaxis()
     
-    [a.grid() for a in [ax1, ax2]]
+    [a.grid() for a in ax]
```
Comment (callumrollo) [inline glidertest/tools.py:506]: looping over one value here?

**Review by callumrollo — COMMENTED** [review]
generally this is a nice pattern of allowing the user to supply an ax if they want to. I've made some suggestions

File: glidertest/tools.py (hunk 3 above)
Comment (MOchiara) [reply glidertest/tools.py:506]: No, it's ax[0] and ax[1]

**Comment by MOchiara** [comment]
@callumrollo shall I do the same for the other plots?

**Comment by callumrollo** [comment]
sure, go for it!

**Pull request merged**